### ISSUE TITLE
drm leasing of the display labwc is using - does wp_drm_lease_v1_destroy restore the desktop?

### ISSUE THREAD

**Issue by jc-kynesim** [labwc/labwc#2137]
I've got some code that uses DRM for display and was surprised/pleased to note the other day that labwc supported leasing of the active display in a similar way to X. I've tried using it and I get the leased handles without issue and I can use them. However when my program ends labwc doesn't come back with a desktop - the screen remains black. I try to shut down gracefully i.e. I use wp_drm_lease_v1_destroy to return my lease. Is there any example or test code where this does work that I can look at?

**Comment by johanmalm** [comment]
Not sure.
I think it's mostly the gaming community that use it.
@Joshua-Ashton @kode54 Any steers in terms of example code?

**Comment by kode54** [comment]
You're more likely to get an answer out of Josh than me. I'd have to run the code to see if it's somehow triggering display loss in wlroots, or if there's a display lease handler we're missing. Does it work in any of the other wlroots compositors, such as sway? I presume you can't test with Wayfire unless you use the track-wlroots branch that follows wlroots-0.18.

**Comment by jc-kynesim** [comment]
labwc was the first compositor that I'd found that supported doing this so I haven't tried any others yet - I was just surprised to find it working at all given that when I looked a while (maybe a year) ago I couldn't find a single compositor that would. This is labwc 0.7.2 (Raspberry Pi). I'll try sway and see what happens. Do happen to know if anything major has changed in this area since 0.17.2?

**Comment by jc-kynesim** [comment]
I can point you at my code (in a bit when I've tidied it slightly) but I make no guarantees on it compiling or working (at all) on anything other than a Pi - it should, but it is completely untested.

**Comment by Consolatis** [comment]
> Does wp_drm_lease_v1_destroy restore the desktop

Yes and no. Based on the docs from wlroots: yes, based on the actual behavior of wlroots: no.
So currently destroying a lease will not restore the desktop which is kind of annoying and labwc will have to do it manually.

See https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4771
